File: java/src/main/be.steinsomers.bron_kerbosch/BronKerboschAlgorithm.java
package be.steinsomers.bron_kerbosch;

public interface BronKerboschAlgorithm {
    int[] EMPTY_CLIQUE = {};

    void explore(UndirectedGraph graph, CliqueConsumer cliqueConsumer) throws InterruptedException;
    default boolean hasRaceCondition() {
        return false;
    }
}

File: kotlin/src/main/be.steinsomers.bron_kerbosch/BronKerboschAlgorithm.kt
package be.steinsomers.bron_kerbosch

@Suppress("SameReturnValue")
interface BronKerboschAlgorithm {
    val name: String
    val deterministic: Boolean
        get() = true

    @Throws(InterruptedException::class)
    fun explore(graph: UndirectedGraph, cliqueConsumer: CliqueConsumer)
}
